### PR TITLE
AP_IntertialNav clarify 2D vs 3D uses and get_bearing_cd() takes Vector2f

### PR DESCRIPTION
**Pull request by hendjoshsr71**
The purpose of this PR is to make the following functions take Vector2f as that is all they use
`get_bearing_cd()`   & `get_horizontal_distance_cm()`

The other purpose is to simplify & clarify places where `inav.position()` is being used, but only in 2D to be `inav.position_xy()` My reasoning for this is I think it will help as an in-between step to remove NEU by making it clearer what is only XY vs Z in later PRs.

It also uses `inav.get_altitude()` or `inav.get_velocity_z()` where appropriate. 

Not sure how useful it is?
This PR saves 380+ bytes on Durandal for copter & just a wee bit elsewhere.

There are a couple spots with PosTypes where attention must be paid to where and how conversion is happening to keep it functionally equivalent.

### CODE REVIEW

**Comment by khancyr** [comment]
we should kill inertial_nav now that the inertial backend are better

File: libraries/AP_InertialNav/AP_InertialNav.cpp [hunk 1]
```diff
@@ -52,34 +52,54 @@ nav_filter_status AP_InertialNav::get_filter_status() const
 }
 
 /**
- * get_position - returns the current position relative to the home location in cm.
+ * get_position_neu_cm - returns the current position relative to the home location in cm.
  *
  * @return
  */
-const Vector3f &AP_InertialNav::get_position(void) const 
+const Vector3f &AP_InertialNav::get_position_neu_cm(void) const
```
Comment (tridge) [inline libraries/AP_InertialNav/AP_InertialNav.cpp:59]: add get_position_z_up_cm()

Comment (hendjoshsr71) [reply libraries/AP_InertialNav/AP_InertialNav.cpp:59]: Fixed

**Comment by hendjoshsr71** [comment]
Renamed to `inav.get_position_z_up_cm`

I also addressed some of the `zzz.length()` spots to use `length_squared()`

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
pending approval from @lthall

**Review by rmackay9 — APPROVED** [review]

**Review by lthall — APPROVED** [review]

File: ArduCopter/mode_follow.cpp [hunk 2]
```diff
@@ -108,9 +108,8 @@ void ModeFollow::run()
         // calculate vehicle heading
         switch (g2.follow.get_yaw_behave()) {
             case AP_Follow::YAW_BEHAVE_FACE_LEAD_VEHICLE: {
-                const Vector3f dist_vec_xy(dist_vec.x, dist_vec.y, 0.0f);
-                if (dist_vec_xy.length() > 1.0f) {
-                    yaw_cd = get_bearing_cd(Vector3f(), dist_vec_xy);
+                if (dist_vec.xy().length() > 1.0f) {
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply ArduCopter/mode_follow.cpp]: Outdated suggestion? Was fixed after approval.

File: ArduCopter/mode_follow.cpp [hunk 3]
```diff
@@ -126,9 +125,8 @@ void ModeFollow::run()
             }
 
             case AP_Follow::YAW_BEHAVE_DIR_OF_FLIGHT: {
-                const Vector3f vel_vec(desired_velocity_neu_cms.x, desired_velocity_neu_cms.y, 0.0f);
-                if (vel_vec.length() > 100.0f) {
-                    yaw_cd = get_bearing_cd(Vector3f(), vel_vec);
+                if (desired_velocity_neu_cms.xy().length() > 100.0f) {
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply ArduCopter/mode_follow.cpp]: Outdated suggestion? Was fixed after approval.

File: libraries/AC_AttitudeControl/AC_PosControl.cpp [hunk 4]
```diff
@@ -404,8 +404,7 @@ float AC_PosControl::pos_offset_z_scaler(float pos_offset_z, float pos_offset_z_
     if (is_zero(pos_offset_z_buffer)) {
         return 1.0;
     }
-    const Vector3f curr_pos = _inav.get_position();
-    float pos_offset_error_z = curr_pos.z - (_pos_target.z - _pos_offset_z + pos_offset_z);
+    float pos_offset_error_z = _inav.get_altitude_cm() - (_pos_target.z - _pos_offset_z + pos_offset_z);
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply libraries/AC_AttitudeControl/AC_PosControl.cpp]: Outdated suggestion? Was fixed after approval.

**Comment by hendjoshsr71** [comment]
@lthall I think your comments were left over from the first DevCall of the week? Each of them had been addressed after your approval.

**Review by lthall — APPROVED** [review]

**Comment by rmackay9** [comment]
I've added the DevCallTopic but that's just so we don't forget about this but we can actually merge any time.

**Comment by rmackay9** [comment]
Sorry to say but this change makes back porting @lthall's recent fixes to 4.1 very difficult..

**Comment by hendjoshsr71** [comment]
Sorry about that. 

I'm not sure if there is a better way to move forward on getting rid ned/neu/cm?

**Pull request merged**